### PR TITLE
feat: ChipSelectField

### PR DESCRIPTION
**Pull request by bdow**
First pass a a ChipSelectField, utilizing our existing Listbox and Option components

### CODE REVIEW

**Review by chr1sjf0x — APPROVED** [review]
Looks great 👍 🍣

File: src/components/internal/Popover.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ interface PopoverProps extends React.PropsWithChildren<any> {
 
 /** Popover is used for generating menus and list-boxes */
 export function Popover(props: PopoverProps) {
```
Comment (KoltonG) [inline src/components/internal/Popover.tsx:13]: **question:** Did we find out if we could get rid of React-Popper and use this? Does this handle changing positions when hitting the edges of the screen?

File: src/inputs/ChipSelectField.test.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    const r = await render(
+      <TestComponent label="Label" value="s:2" options={sports} onFocus={onFocus} onBlur={onBlur} />,
+    );
+
+    // When focusing the field and opening the menu
+    click(r.chipSelectField);
+    // And onFocus should have been called
+    expect(onFocus).toBeCalledTimes(1);
+    // And when firing the blur event while the menu is opened,
+    fireEvent.blur(r.chipSelectField());
```
Comment (KoltonG) [inline src/inputs/ChipSelectField.test.tsx:94]: **question:** This is neat! How are you controlling the blur to not be propagated?

File: src/inputs/ChipSelectField.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    <>
+      <div
+        css={{
+          ...chipStyles,
+          ...Css.dif.p0.mwPx(32).if(!value).bgGray200.$,
+          ...(isFocused ? Css.bshFocus.$ : {}),
+        }}
+      >
+        <Label label={label} labelProps={labelProps} hidden {...tid.label} />
+        <button
```
Comment (KoltonG) [inline src/inputs/ChipSelectField.tsx:141]: Is this the rounded button?

File: src/inputs/ChipSelectField.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    shouldFlip: true,
+    scrollRef: listBoxRef,
+    isOpen: state.isOpen,
+    onClose: state.close,
+    placement: "bottom left",
+  });
+
+  positionProps.style = {
+    ...positionProps.style,
+    width: buttonRef?.current?.clientWidth,
```
Comment (KoltonG) [inline src/inputs/ChipSelectField.tsx:126]: Nice, this matches the button width!

File: src/inputs/ChipSelectField.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      {state.isOpen && (
+        <Popover
+          triggerRef={buttonRef}
+          popoverRef={popoverRef}
+          positionProps={positionProps}
+          onClose={state.close}
+          isOpen={state.isOpen}
+          shouldCloseOnBlur={true}
+        >
+          <ListBox
```
Comment (KoltonG) [inline src/inputs/ChipSelectField.tsx:182]: **question:** Is this the virtual one?

**Review by KoltonG — APPROVED** [review]
🍣

File: src/inputs/ChipSelectField.tsx (hunk 5 above)
Comment (bdow) [reply src/inputs/ChipSelectField.tsx:182]: Yes, this is the virtualized one.

File: src/inputs/ChipSelectField.tsx (hunk 3 above)
Comment (bdow) [reply src/inputs/ChipSelectField.tsx:141]: Yes. This button matches the designs for the Chips with just a different color to follow the SchedulesV2 design. This is a prototype component, so not forcing it to comply with existing styles (but it does for the most part).

File: src/inputs/ChipSelectField.test.tsx (hunk 2 above)
Comment (bdow) [reply src/inputs/ChipSelectField.test.tsx:94]: We are able to conditionally call on blur based on whether the menu is open or not.

File: src/components/internal/Popover.tsx (hunk 1 above)
Comment (bdow) [reply src/components/internal/Popover.tsx:13]: Yes, this does handle changing positions. We could get rid of React-Popper, but we might want to generalize this component a bit more before then. Part of some tech debt.

File: src/inputs/ChipSelectField.test.tsx (hunk 2 above)
Comment (KoltonG) [reply src/inputs/ChipSelectField.test.tsx:94]: Is that through React-Aria?

File: src/components/internal/Popover.tsx (hunk 1 above)
Comment (KoltonG) [reply src/components/internal/Popover.tsx:13]: Nice! Great find!

File: src/inputs/ChipSelectField.test.tsx (hunk 2 above)
Comment (bdow) [reply src/inputs/ChipSelectField.test.tsx:94]: Not aria, but, React-Stately - same idea. We know when the menu is opened/closed.

**Pull request merged**